packages/lit-dev-content/site/docs/composition/component-composition.md

```diff
@@ -0,0 +1,110 @@
+
+For more information on dispatching and listening for events, see [Events](/docs/components/events/).
+
+
+## Passing data across the tree
+
+Properties down and events up is a good rule to start with. But what if you need to exchange data between two components that don't have a direct descendant relationship? For example, two components that are siblings in the shadow tree?
+
+One solution to this problem is to use the _mediator pattern_. In the mediator pattern, peer components don't communicate with each other directly. Instead, interactions are _mediated_ by a third party.
+
+A simple way to implement the mediator pattern is by having the owning component handle events from its children, and in turn update the state of its children as necessary by passing changed data back down the tree. By adding a mediator, you can pass data across the tree using the familiar events-up, properties-down principle.
+
+In the following example, the mediator element listens for events from the input and button elements in its shadow DOM. It controls the enabled state of the button so the user can only click **Submit** when there's text in the input.
+
+{% playground-example "docs/composition/mediator-pattern" "mediator-element.ts" %}
+
+Other mediator patterns include flux/Redux-style patterns where a store mediates changes and updates components via subscriptions. Having components directly subscribe to changes can help avoid needing every parent to pass along all data required by its children.
+
+## Light DOM children
+
+In addition to the nodes in your shadow DOM, you can render child nodes provided by the component user, like the standard `<select>` element can take a set of `<option>` elements as children and render them as menu items.
+
+Child nodes are sometimes referred to as "light DOM" to distinguish them from the component's shadow DOM. For example:
+
+```html
+<top-bar>
+  <icon-button icon="menu" slot="nav-button"></icon-button>
+  <span slot="title">Fuzzy</span>
+</top-bar>
+```
+
+Here the `top-bar` element has two light DOM children supplied by the user: a navigation button, and a title.
+
+Interacting with light DOM children is different from interacting with nodes in the shadow DOM. Nodes in a component's shadow DOM are managed by the component, and shouldn't be accessed from outside the component. Light DOM children are managed from outside the component, but can be accessed by the component as well. The component's user can add or remove light DOM children at any time, so the component can't assume a static set of child nodes.
+
+The component has control over whether and where the child nodes are rendered, using the `<slot>` element in its shadow DOM. And it can receive notifications when child nodes are added and removed by listening for the `slotchange` event.
```

Comment:
could add a TODO/issue here about interacting with light DOM children: listening to events (where to install the listener, how to clean up if you installed them directly on the children and they are removed), maybe mentioning MutationObserver and that `slotchange` is better because of reprojection, discuss when or whether to set attributes and properties, etc.

Registration order and events are a tricky topic that should be covered somewhere - it's possible when trying to use events to communicate that a child can fire them before the parent is upgraded and can listen.

Reply:
Added #628 to track this.